### PR TITLE
feat: add development Dockerfile and rename existing docker file.

### PR DESCRIPTION
**Pull request by iwknow**
## Motivation

Developers want to have a more convenient development docker image for sglang-jax.
 
## Modifications
- re-organize the `Dockerfiles` under `docker` directory.
- rename the existing `Dockerfile` to `Dockerfile.runtime`.
- add `Dockerfile.dev` for development image.
- add `README` for a brief explanation. 
- 
## Accuracy Tests

Build the image manually without any issue.

## Benchmarking and Profiling

N/A

## Checklist

- [X] Please use English, otherwise it will be closed.
- [X] The purpose of the PR, or link existing issues this PR will resolve.
- [X] The test plan, such as providing test command.
- [ ] (Optional) The necessary documentation update.

### CODE REVIEW

**Review by cjx0709 — APPROVED** [review]
LGTM
